emukit/core/optimization/acquisition_optimizer.py

```diff
@@ -26,14 +27,38 @@ def optimize(self, acquisition: Acquisition, context: dict = None) -> Tuple[np.n
 
         :param acquisition: The acquisition function to be optimized
         :param context: Optimization context, determines whether any variable values should be fixed during the optimization
+        :return: Tuple of (optimum value, location of optimum)
         """
 
-        self.gpyopt_acquisition_optimizer.context_manager = GPyOpt.optimization.acquisition_optimizer.ContextManager(
-            self.gpyopt_space, context)
-
         # Take negative of acquisition function because they are to be maximised and the optimizers minimise
         f = lambda x: -acquisition.evaluate(x)
 
+        # Validate context
```

Comment:
This method becomes a bit lengthy. Can you create smaller functions to move some code in there and reduce the complexity? An easy one: `_validate_context`

Reply:
Will do